tutorials/mpi-fortran/code/reduce_avg.f90

```diff
@@ -0,0 +1,78 @@
+module subs
+  implicit none
+contains
+  subroutine create_rand_nums(rand_nums, num_elements)
+    ! Creates an array of random numbers. Each number has a value from 0 - 1
+    integer, intent(in) :: num_elements
+    real, intent(out)   :: rand_nums(num_elements)
+
+    integer :: i
+
+    do i = 1, num_elements
+      rand_nums(i) = rand()
+    end do
+
+  end subroutine create_rand_nums
+end module subs
+
+program main
+  use mpi_f08
+  use iso_fortran_env, only: error_unit
+  use subs
+
+  implicit none
+
+  integer :: num_args
+  character(12) :: arg
+  integer :: num_elements_per_proc
+  integer :: world_size, world_rank
+  real :: r, local_sum, global_sum
+  real, allocatable :: rand_nums(:)
+
+  num_args = command_argument_count()
+
+  if (num_args .ne. 1) then
+    write (error_unit, *) 'Usage: reduce_avg num_elements_per_proc'
+    stop
+  end if
+
+  call get_command_argument(1, arg)
+  read (arg, *) num_elements_per_proc
+
+  call MPI_INIT()
+
+  call MPI_COMM_SIZE(MPI_COMM_WORLD, world_size)
+  call MPI_COMM_RANK(MPI_COMM_WORLD, world_rank)
+
+  ! Create a random array of elements on all processes.
+  call srand(time()) ! Seed the random number generator to get different results each time for each processor
+  ! Throw away first rand value
```

Comment:
Explain why?

Reply:
After `srand` the first call to `rand` always returns a large number (it looks like the seed used), so I was throwing this away.  This issue has gone away when using `random_seed` and `random_number`.